src/nccl_ofi_rdma.c

```diff
@@ -5503,9 +5468,24 @@ static int connect(nccl_net_ofi_ep_t *base_ep,
 
 		comm_state->stage = COMM_CONN_RESP_REQ_PENDING;
 
+	case COMM_CONN_RESP_REQ_PENDING:
+
+		// TODO: We might want to use READ_ONCE to read variable `connected'
```

Comment:
I think now we can solve this TODO by simply calling `finish_connect` within connect itself. That avoids the problem of another thread potentially calling `finish_connect`.